amount/main.go

```
@@ -84,8 +90,22 @@ func ParseInt64(v string) (int64, error) {
 // "amount". In other words, it divides the given string integer value by 10^7
 // and returns the string representation of that number.
 // It is safe to use with values exceeding int64 limits.
-func IntStringToAmount(v string) (string, error) {
-	if !validAmountSimple.MatchString(v) {
+//
+// Pass AllowOverInt64 as an option to allow v values over int64. Please note
+// that computations may take much more time.
+//
+// Note: this can be rewritten to not use big.Rat. We should do it if there are
+// serious performance issues.
+func IntStringToAmount(v string, opts ...Option) (string, error) {
+	allowOverInt64 := false
+	for _, opt := range opts {
+		switch opt {
+		case AllowOverInt64:
+			allowOverInt64 = true
+		}
+	}
+
+	if !allowOverInt64 && !validAmountSimple.MatchString(v) {
```

@bartekn where in the code do we _fix_ the bug that causes the error?

my understanding is that now we will correctly catch and throw the error, but the error still exists.
I did see the new test and may have missed the part of the PR that fixes it so want to make sure it's there.

---

The actual fix is here: https://github.com/stellar/go/pull/609/files#diff-981180cdf0abda8125a97a2d79bcb503L24

https://github.com/stellar/go/commit/5bf53ca1550b70f00e6294589d4fcb2497264b31 was a result of a security report submitted to SDF's Bug Bounty program. I changed the `amount` package to return an error if the amount passed as an argument is too large. This was a security fix because passing a parameter to `/paths` like: `/paths?destination_amount=1E9223372036854775807` could basically DoS Horizon. I didn't realize (and also we don't have tests that cover this) that `amount.IntStringToAmount` function is used in `/assets` and sum of balances of an asset can be larger than `int64`. The line I linked above was basically failing silently without copying the rest of fields from DB result object. The change was to allow values larger that `int64` in `amount.IntStringToAmount` by passing an option: `AllowOverInt64`.

`amount.IntStringToAmount` function can be changed to remove `big.Rat` dependency (simple string algorithm). I'm planning to do it tomorrow and deploy patch releases (0.13.3 and 0.14.0rc3).

---

ok I see, makes sense. thanks!